elements/pf-progress/pf-progress.ts

```diff
@@ -0,0 +1,228 @@
+ *          Background color of the progress bar when variant is success.
+ *          {@default `#3e8635`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-warning__bar--BackgroundColor
+ *          Background color of the progress bar when variant is warning.
+ *          {@default `#f0ab00`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-danger__bar--BackgroundColor
+ *          Background color of the progress bar when variant is danger.
+ *          {@default `#c9190b`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-success__status-icon--Color
+ *          Color of the status icon when variant is success.
+ *          {@default `#3e8635`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-warning__status-icon--Color
+ *          Color of the status icon when variant is warning.
+ *          {@default `#f0ab00`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-danger__status-icon--Color
+ *          Color of the status icon when variant is danger.
+ *          {@default `#c9190b`}
+ *
+ * @cssprop {<length>} --pf-c-progress--m-inside__indicator--MinWidth
+ *          Minimum width of the progress bar indicator when measure location is inside.
+ *          {@default `2rem`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-inside__measure--Color
+ *          Color of the progress bar measure when measure location is inside.
+ *          {@default `#ffffff`}
+ *
+ * @cssprop {<color>} --pf-c-progress--m-success--m-inside__measure--Color
+ *          Color of the progress bar measure when variant is success and measure location is inside.
+ *          {@default `#ffffff`}
+ *
+ * @cssprop {<length>} --pf-c-progress--m-inside__measure--FontSize
+ *          Font size of the progress bar measure when measure location is inside.
+ *          {@default `0.875rem`}
+ *
+ * @cssprop {<length>} --pf-c-progress--m-outside__measure--FontSize
+ *          Font size of the progress bar measure when measure location is outside.
+ *          {@default `0.875rem`}
+ *
+ * @cssprop {<length>} --pf-c-progress--m-sm__bar--Height
+ *          Height of the progress bar when the size is small.
+ *          {@default `0.5rem`}
+ *
+ * @cssprop {<length>} --pf-c-progress--m-sm__description--FontSize
+ *          Font size of the progress bar description when the size is small.
+ *          {@default `0.875rem`}
+ *
+ * @cssprop {<length>} --pf-c-progress--m-sm__measure--FontSize
+ *          Font size of the progress bar measure when the size is small.
+ *          {@default `0.875rem`}
+ *
+ * @cssprop {<length>} --pf-c-progress--m-lg__bar--Height
+ *          Height of the progress bar when the size is large.
+ *          {@default `1.5rem`}
+ *
+ */
+@customElement('pf-progress')
+export class PfProgress extends LitElement {
+  static readonly styles = [styles];
+
+  static readonly formAssociated = true;
+
+  #calculatedPercentage = 0;
+
+  #logger = new Logger(this);
+
+  @property({ reflect: true, type: Number })
+    value = 0;
+
+  @property()
+    title = '';
```

Comment:
Should `title` be a slotted element instead?  

I realize this is most likely mocking PFv4 API, but the title attribute is a [global html attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/title) so this feels like it is overloading that with a new implementation.